subscription.go

```diff
@@ -17,35 +17,36 @@ const (
-	PaidThroughDate         string               `xml:"paid-through-date"`
-	PaymentMethodToken      string               `xml:"payment-method-token"`
-	PlanId                  string               `xml:"plan-id"`
-	Price                   *Decimal             `xml:"price"`
-	Status                  SubscriptionStatus   `xml:"status"`
-	TrialDuration           string               `xml:"trial-duration"`
-	TrialDurationUnit       string               `xml:"trial-duration-unit"`
-	TrialPeriod             bool                 `xml:"trial-period"`
-	Transactions            *Transactions        `xml:"transactions"`
-	Options                 *SubscriptionOptions `xml:"options"`
-	Descriptor              *Descriptor          `xml:"descriptor"`
-	AddOns                  *AddOnList           `xml:"add-ons"`
-	Discounts               *DiscountList        `xml:"discounts"`
+	XMLName                 string                    `xml:"subscription"`
+	Id                      string                    `xml:"id"`
+	Balance                 *Decimal                  `xml:"balance"`
+	BillingDayOfMonth       string                    `xml:"billing-day-of-month"`
+	BillingPeriodEndDate    string                    `xml:"billing-period-end-date"`
+	BillingPeriodStartDate  string                    `xml:"billing-period-start-date"`
+	CurrentBillingCycle     string                    `xml:"current-billing-cycle"`
+	DaysPastDue             string                    `xml:"days-past-due"`
+	FailureCount            string                    `xml:"failure-count"`
+	FirstBillingDate        string                    `xml:"first-billing-date"`
+	MerchantAccountId       string                    `xml:"merchant-account-id"`
+	NeverExpires            bool                      `xml:"never-expires"`
+	NextBillAmount          *Decimal                  `xml:"next-bill-amount"`
+	NextBillingPeriodAmount *Decimal                  `xml:"next-billing-period-amount"`
+	NextBillingDate         string                    `xml:"next-billing-date"`
+	NumberOfBillingCycles   *int                      `xml:"number-of-billing-cycles"`
+	PaidThroughDate         string                    `xml:"paid-through-date"`
+	PaymentMethodToken      string                    `xml:"payment-method-token"`
+	PlanId                  string                    `xml:"plan-id"`
+	Price                   *Decimal                  `xml:"price"`
+	Status                  SubscriptionStatus        `xml:"status"`
+	TrialDuration           string                    `xml:"trial-duration"`
+	TrialDurationUnit       string                    `xml:"trial-duration-unit"`
+	TrialPeriod             bool                      `xml:"trial-period"`
+	Transactions            *Transactions             `xml:"transactions"`
+	Options                 *SubscriptionOptions      `xml:"options"`
+	StatusHistory           []SubscriptionStatusEvent `xml:"status-history>status-event"`
```

Comment:
I think this reference should still be a pointer for consistency with the other types in the SDK. I agree with your point that the pointer is unnecessary, but at this point the consistency is worth it I think.